pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -621,7 +627,12 @@ func (r *reconcilerBase) setupOrTeardown(ctx context.Context, syncObj client.Obj
 
 	if controllerutil.ContainsFinalizer(syncObj, metadata.ReconcilerManagerFinalizer) {
 		// Our finalizer is present, so delete managed resource objects.
-		if err := teardownFn(ctx); err != nil {
+		var nre *NoRetryError
+		if err := teardownFn(ctx); errors.As(err, &nre) {
```

Comment:
nit: might be slightly easier to read as:
```
		if err := teardownFn(ctx); err != nil {
			if errors.As(err, &nre) {
				r.logger(ctx).Error(err, "Teardown failed (waiting for resource status to change)")
				// Return nil to avoid triggering immediate retry.
				return nil
			}
			return errors.Wrap(err, "teardown failed")
		}
```

Reply:
Agreed, done